### PR TITLE
Update Module::getPaymentModules()

### PR DESCRIPTION
**Pull request by hekep**
Need possibility to provide $context as parameter.

There are custom situations, when customer is not logged in, and I need to create $context from scratch, for example  AnyPaymentValidationModuleFrontController :

$authorized = false;
$myContext = Context::getContext();
$myContext->cart = new Cart(Tools::getValue('id_cart'));
		foreach (Module::getPaymentModules($myContext ) as $module)
			if ($module['name'] == 'myPayment')
			{
				$authorized = true;
				break;
			}
		if (!$authorized)
			die($this->module->getTranslator()->trans('This payment method is not available.', array(), 'Modules.Wirepayment.Shop'));

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop / 1.7.6.x / 1.6.1.x
| Description?  | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc.
| Type?         | bug fix / improvement / new feature / refacto / critical
| Category?     | FO / BO / CO / IN / WS / TE
| BC breaks?    | Does it break backward compatibility? yes/no
| Deprecations? | Does it deprecate an existing feature? yes/no
| Fixed ticket? | [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Please indicate how to best verify that this PR is correct.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16388)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Hi and thanks for your contribution.
Could you please fill the table with all needed information?
Could you check for the CI? (PrettyCi & Travis)

Regards

File: classes/module/Module.php [hunk 1]
```diff
@@ -1922,9 +1922,11 @@ public static function getModulesInstalled($position = 0)
...
      * @return array module informations
      */
-    public static function getPaymentModules()
+    public static function getPaymentModules($context = null)
     {
-        $context = Context::getContext();
+        if(!$context){
+            $context = Context::getContext();
+        }
         if (isset($context->cart)) {
```
Comment (hekep) [inline classes/module/Module.php:1930]: PrettyCi , now what should I do`?

**Review by hekep — COMMENTED** [review]
Questions | Answers
-- | --
Branch  1.7.6.x 

Type  : improvement / new feature / refacto 
Category: | FO / BO / CO / IN / WS / TE
BC breaks: no
Deprecations? | Does it deprecate an existing feature? yes/no
How to test?

**Comment by hekep** [comment]
I get M2M rest calls to  AnyPaymentValidationModuleFrontController , so customer is not logged in and   Context::getContext(); is to dumb. I trust the M2M call, recreate $context in my AnyPaymentValidationModuleFrontController ...

**Comment by hekep** [comment]
I have no idea , what am I suppose to do.

**Comment by PierreRambaud** [comment]
@hekep You need to complete the description of this pull request, following the table.
Example: https://github.com/PrestaShop/PrestaShop/pull/16489#issue-343941477
You also need to fix your code to follow our standard: https://github.com/PrestaShop/PrestaShop/pull/16388/checks?check_run_id=300744376
There is a command to do it automatically: `./vendor/bin/php-cs-fixer fix`

**Comment by PierreRambaud** [comment]
HI @hekep 
Since we had no news from you for more than 30 days, I'll close this pull request

Thanks!

**Pull request closed without merging**